### ISSUE TITLE
scaling traefic example in swarm mode

### ISSUE THREAD

**Issue by dcrystalj** [containous/traefik#1200]
It would be really nice if you could provide example script which shows running multiple traefik instances in cluster, to avoid single point of failure.

I know cluster needs multiple consul instances to share config file for all traefic containers, but from documentation it is very hard to get.

**Comment by CyrilPeponnet** [comment]
This is how I've done it with docker swarm if it can help.

```yaml
version: "3"
services:
  traefik:
    image: cyrilpeponnet/traefik:latest
    deploy:
      update_config:
        parallelism: 1
        delay: 10s
      mode: global
      placement:
        constraints: [node.role == manager]
    environment:
      - "AWS_ACCESS_KEY_ID=BLA"
      - "AWS_SECRET_ACCESS_KEY=BLABLA"
      - "AWS_REGION=us-west-2"
    ports:
      - 80:80
      - 443:443
      - 8080:8080
    networks:
      - backend
      - consul-net
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    command: |-
      -c /dev/null
      --web
      --consul
      --consul.endpoint=consul-cluster:8500
      --consul.watch=true
      --docker
      --docker.swarmmode
      --docker.domain=traefik
      --docker.watch
      --debug

  consul-cluster:
    image: consul:latest
    deploy:
      update_config:
        parallelism: 1
        delay: 10s
      mode: global
      placement:
        constraints: [node.role == manager]
    ports:
      - 8500:8500
    networks:
      - consul-net
    environment:
      - 'CONSUL_BIND_INTERFACE=eth2'
    command: agent -server -datacenter traefik -ui -retry-join=tasks.consul-cluster -retry-join=tasks.consul-cluster -retry-join=tasks.consul-cluster -bootstrap-expect=3 -client=0.0.0.0


networks:
  backend:
      driver: overlay
  consul-net:
      driver: overlay
```

**Comment by dcrystalj** [comment]
thank you sir!  This is what I was looking for, but for some reason it doesn't spin the services using `deploy` command. I get `task: non-zero exit (255)"` for both traefik and consul.

**Comment by llitfkitfk** [comment]
1. compose.yml:

![image](https://user-images.githubusercontent.com/2404785/29311378-59e668a8-81e4-11e7-8a0b-e9ce71f79320.png)

![image](https://user-images.githubusercontent.com/2404785/29311360-44ef58e2-81e4-11e7-9ffa-a43895ea9703.png)

2. `docker stack deploy -c compose.yml gateway`  to deploy traefik & whoami :

![image](https://user-images.githubusercontent.com/2404785/29311559-003dfe32-81e5-11e7-814a-fe33ad7bd109.png)

3. `docker service scale gateway_whami=10`,  dashboard no change:

![image](https://user-images.githubusercontent.com/2404785/29311559-003dfe32-81e5-11e7-814a-fe33ad7bd109.png)

**Comment by traefiker** [comment]
Closed by #2589.
